[BUG] provider_wit_bindgen::generate! crate paths and `encode_sync` error for funcs with result<> return type

### Affected project(s)

- [ ] documentation
- [ ] examples
- [ ] wasmCloud host
- [ ] wasmCloud CLI (wash)
- [ ] wasmCloud dashboard UI (washboard)
- [ ] capability providers
- [X] provider bindgen
- [ ] control interface client
- [ ] other / not sure

### Describe the bug

There seems to be a path in `wasmcloud_provider::wit_bindgen` that is choking on the following wit:

```
interface types {
  record quaternion {
    x: f32,
    y: f32,
    z: f32,
    w: f32,
  }
  variant access-error {
    named-resource-not-found(string),
    hardware-access-error(string)
  }
}

interface foo {
    use types.{
        quaternion, access-error
    };
    schmoo: func(q: quaternion) -> result<quaternion, access-error>;
}
```

afaict there are a few issues:

1. compiling a provider lib with wasmcloud_provider_wit_bindgen::generate! and that wit results in an error:
error[E0433]: failed to resolve: use of undeclared crate or module `wrpc_transport_derive`
  --> foo/src/lib.rs:14:1
   |
14 | / wasmcloud_provider_wit_bindgen::generate!({
15 | |     impl_struct: FooProvider,
16 | |     contract: "brpc:examples",
17 | |     wit_bindgen_cfg: "imu-host-component"
18 | | });
   | |__^ use of undeclared crate or module `wrpc_transport_derive` 

which I think can be traced to: https://github.com/wasmCloud/wasmCloud/blob/main/crates/wrpc-transport-derive-macro/src/wrpc_transport/receive.rs#L206 - replacing that with 
Current:
```rust
                let (discriminant, payload) = wrpc_transport::receive_discriminant(payload, rx)

```

New:
```rust
                let (discriminant, payload) = #crate_path::deps::wrpc_transport::receive_discriminant(payload, rx)
``` 

that resolves the first issue.  We then get:

```sh
error[E0599]: no method named `encode_sync` found for enum `std::result::Result` in the current scope
  --> foo/src/lib.rs:15:1
   |
15 |   wasmcloud_provider_wit_bindgen::generate!({
   |  _^
16 | |     impl_struct: FooProvider,
17 | |     contract: "brpc:examples",
18 | |     wit_bindgen_cfg: "imu-host-component"
19 | | });
   | |__^ method not found in `Result<Quaternion, AccessError>`
   |
   = note: this error originates in the macro `wasmcloud_provider_wit_bindgen::generate` (in Nightly builds, run with -Z macro-backtrace for more info)
```

### Steps to reproduce

define a .wit function returning a `result<some-struct, some-error>`

cargo build

### Expected behavior

no error emitted when using the

wasmcloud_provider_wit_bindgen::generate!({
    impl_struct: FooProvider,
    contract: "brpc:examples",
    wit_bindgen_cfg: "imu-host-component"
});

macro

### Environment

- OS: OS X & Linux
- Shell : zsh
- wasmcloud Version: main branch


### Screenshots / Logs / Additional context

_No response_

--- a/crates/wrpc-transport-derive-macro/src/wrpc_transport/receive.rs
+++ b/crates/wrpc-transport-derive-macro/src/wrpc_transport/receive.rs
@@ -203,7 +203,8 @@ fn derive_receive_inner_for_enum(item: syn::Item) -> Result<TokenStream> {
             where
                 T: #crate_path::deps::futures::Stream<Item=#crate_path::deps::anyhow::Result<#crate_path::deps::bytes::Bytes>> + Send + Sync + Unpin + 'static
             {
-                let (discriminant, payload) = wrpc_transport_derive::deps::wrpc_transport::receive_discriminant(payload, rx)
+                use #crate_path::deps::anyhow::Context as _;
+                let (discriminant, payload) = #crate_path::deps::wrpc_transport::receive_discriminant(payload, rx)
                     .await
                     .context("failed to receive enum discriminant")?;
                 match discriminant {

